### PR TITLE
Update qr_code_scanner.dart fix widgetsBinding no longer nullable for Flutter 3

### PR DESCRIPTION
**Pull request by MoaidAlrazhy**

### CODE REVIEW

File: lib/src/qr_code_scanner.dart [hunk 1]
```diff
@@ -65,7 +65,7 @@ class _QRViewState extends State<QRView> {
   void initState() {
     super.initState();
     _observer = LifecycleEventHandler(resumeCallBack: updateDimensions);
-    WidgetsBinding.instance!.addObserver(_observer);
+    WidgetsBinding.instance.addObserver(_observer);
```
Comment (xieenming) [inline lib/src/qr_code_scanner.dart:68]: Should change to `WidgetsBinding.instance?.addObserver(_observer);`

File: lib/src/qr_code_scanner.dart [hunk 2]
```diff
@@ -83,7 +83,7 @@ class _QRViewState extends State<QRView> {
   @override
   void dispose() {
     super.dispose();
-    WidgetsBinding.instance!.removeObserver(_observer);
+    WidgetsBinding.instance.removeObserver(_observer);
```
Comment (xieenming) [inline lib/src/qr_code_scanner.dart:86]: Should change to `WidgetsBinding.instance?.removeObserver(_observer);`

**Review by xieenming — CHANGES_REQUESTED** [review]
Need to change the commit

**Comment by MoaidAlrazhy** [comment]
its no longer nullable so no need for null check

**Comment by LowProgram** [comment]
Completly agree with @MoaidAlrazhy, they simplified the API for Flutter 3. So I've made these changes as well and ran into the following errors.  
image
Anybody an idea how to solve those?

**Comment by juliansteenbakker** [comment]
For flutter v3 this is fine. Version 1.0.0 of this package will only be supported with flutter v3.

**Pull request merged**